### ISSUE TITLE
Entity:SetNWEntity() causes Entity:SetNWBool() to fail

### ISSUE THREAD

**Issue by EstevanTH** [Facepunch/garrysmod-issues#1900]
Hi Rubat, William, etc. :heart:

I found out that in Garry's Mod 15, **Entity:SetNWEntity()** causes **Entity:SetNWBool()** to fail.
My bug has been tested on the *2015.03.16* dev version.

Here is a Lua code example, I invite you to autorun it.
```lua
local Entity = FindMetaTable( "Entity" )
local old_SetKeyValue = Entity.SetKeyValue
function Entity:SetKeyValue( key, value, ... )
	self:SetNWEntity( 'MyDebugVehicleVehicle', self ) -- can be any entity
	self:SetNWBool( 'MyDebugVehicleScript', true )
	return old_SetKeyValue( self, key, value, ... )
end
```

Now spawn a Jeep or any `prop_vehicle_jeep*` vehicle.
Run this in your console:
```lua
lua_run print(ents.FindByClass('prop_vehicle_jeep*')[1]:GetNWBool('MyDebugVehicleScript')); lua_run_cl print(ents.FindByClass('prop_vehicle_jeep*')[1]:GetNWBool('MyDebugVehicleScript'))
```

Remove your first vehicle in order to have the index `[1]` on your new vehicle.
Now spawn again the same vehicle (or another one?).
Run the same command in your console.
**Entity:GetNWBool()** does not work anymore because of **Entity:SetNWBool()**.

P.S.: Well Rubat, in the *2015.03.16* dev version **DLabel:SetFGColor()** does not work anymore. I count on you to find out what is wrong.

P.S.: My bus completely failed with GMod 15 updates. :sob:

**Comment by robotboy655** [comment]
```P.S.: Well Rubat, in the 2015.03.16 dev version DLabel:SetFGColor() does not work anymore. I count on you to find out what is wrong.```

This is how it was before the big update this year. You can verify this by using this version of DLabel:
https://github.com/garrynewman/garrysmod/commit/d79de3af05f1b2069fa6e6a1a144d63fc5311ba2

Use SetTextColor

**Comment by EstevanTH** [comment]
Okay Rubat.

I just tested a way to solve the issue before any update.
**Entity:SetNWInt()**, **Entity:SetNWFloat()** and even **Entity:SetNWString()** do not work better than **Entity:SetNWEntity()**.

Using only NWStrings everywhere does not make it work better.

I tried many things and nothing works. I'm so sorry that I could not know that a new version of GMod would be released because I know no place to get this piece of information. Otherwise I would have warned you before the GMod 15 release. :worried:

I hope that you will find out why there are such problems. I want to be confident with Garry's Mod features... :sweat_smile: (Because making my own Lua networking system is not a good solution.)

Love,

Momo.

**Comment by thegrb93** [comment]
Confirmed

**Comment by EstevanTH** [comment]
No, I'm using a `timer(0, function () ..... end)` on the vehicle, and I network my stuff there. And my vehicle is not an SENT.
Maybe it only affects `prop_vehicle*` entities or something like that. I have no time to test.

**Comment by thegrb93** [comment]
I just used myself. I can try a vehicle.

**Comment by EstevanTH** [comment]
Remember: it will work with your 1st vehicle, but it won't with your next vehicles.
As you can't create a removed player entity, I guess that your test was incorrect.

**Comment by thegrb93** [comment]
Server
```
> Entity(112):SetNWEntity("test", Entity(112)) Entity(112):SetNWBool("test2",true)...

> print(Entity(112):GetNWEntity("test"), Entity(112):GetNWBool("test2"))...
Vehicle [112][prop_vehicle_jeep_old]	true


> Entity(113):SetNWEntity("test", Entity(113)) Entity(113):SetNWBool("test2",true)...

> print(Entity(113):GetNWEntity("test"), Entity(113):GetNWBool("test2"))...
Vehicle [113][prop_vehicle_jeep_old]	true
```
Client
```
> print(Entity(112):GetNWEntity("test"), Entity(112):GetNWBool("test2"))...
Vehicle [112][prop_vehicle_jeep_old]	true


> print(Entity(113):GetNWEntity("test"), Entity(113):GetNWBool("test2"))...
Vehicle [113][prop_vehicle_jeep_old]	true
```
The entities had the same index on the server and client, I checked.

**Comment by EstevanTH** [comment]
Okay I will try with a 100% clear installation of Garry's Mod 15.
I do not think I will find the actual reason of the failure.

Garry's Mod 15 was designed to make my vehicles fail. :innocent: As mother nature is nasty, I will make a system compatible with `net`, `umsg` and `BroadcastLua()`, the networking way will be selected with a setting. I will never care again of how Garry's Mod works: failure => change setting. :sunglasses: I did not say it, but I am tired of failing updates. :tired_face:

**Comment by EstevanTH** [comment]
It seems that I cannot find out what is wrong.
A clean installation made me able to use my bus correctly again, so there may be something that has not been correctly cleaned up during the GMod 15 migration.

**Comment by EstevanTH** [comment]
The bus is solved in Garry's Mod **2015.25.03** (what a strange version number by the way).
I am closing the bug report now.

**Comment by EstevanTH** [comment]
Hi everyone.
I have exactly the same problem. I have not tried debugging again, but NWVars fail with my bus exactly the same way.
Rubat, have you tested [my bus](http://steamcommunity.com/sharedfiles/filedetails/?id=385180265) with the new NWVars as an example?

**Comment by EstevanTH** [comment]
Here are the conditions where I have the bug:
- New game started from the Garry's Mod client
- 2 players max or more

The bug seems not to happen in single player and with dedicated servers.

**Comment by thegrb93** [comment]
I tried again using a client server.

```
] lua_run print(Entity(82))
Vehicle [82][prop_vehicle_jeep_old]

] lua_run Entity(82):SetNWBool("Hi",true) Entity(82):SetNWEntity("HiEnt",Entity(1))

] lua_run print(Entity(82):GetNWBool("Hi"), Entity(82):GetNWEntity("HiEnt"))
true	Player [1][-|Sparky|-]

] lua_run_cl print(Entity(82):GetNWBool("Hi"), Entity(82):GetNWEntity("HiEnt"))
true	Player [1][-|Sparky|-]
```

**Comment by EstevanTH** [comment]
Rubat, I do not have the bug since the last update, that's why I just closed it.

**Comment by Bo98** [comment]
Because the new NWVars were reverted. They will probably come back in the future and this should be fixed before then.

**Comment by EstevanTH** [comment]
Reverted again? I did not see that in the changelog!
Rubat, take your time with this feature, I'm not sure that people will be happy while it does not work fine. And it is not such a big feature. :smile:
